lib/puppet/pops/evaluator/closure.rb

```diff
@@ -56,7 +55,7 @@ def call_by_name(args_hash, enforce_parameters)
               # No default. Assign nil if the args_hash included it
               param_scope[name] = nil if args_hash.include?(name)
             else
-              param_scope[name] = param_scope.evaluate(name, p.value, @enclosing_scope, @evaluator)
+              param_scope[name] = param_scope.evaluate(name, p.value, enclosing_scope, @evaluator)
```

Comment:
Given that #enclosing_scope now might be a full lookup, it should only take place once in #call_by_name and then be accessed from a local variable.